clients/hierarchy/hierarchy.go

```diff
@@ -67,38 +75,46 @@ func (c *Client) Healthcheck() (string, error) {
 }
 
 // GetRoot returns the root hierarchy response from the hierarchy API
-func (c *Client) GetRoot(instanceID, name string) (m Model, err error) {
+func (c *Client) GetRoot(ctx context.Context, instanceID, name string) (m Model, err error) {
 	path := fmt.Sprintf("/hierarchies/%s/%s", instanceID, name)
 
-	clientlog.Do(context.Background(), "retrieving hierarchy", service, path, log.Data{
+	clientlog.Do(ctx, "retrieving hierarchy", service, path, log.Data{
 		"method":      "GET",
 		"instance_id": instanceID,
 		"dimension":   name,
 	})
 
-	return c.getHierarchy(path)
+	return c.getHierarchy(path, ctx)
 }
 
 // GetChild returns a child of a given hierarchy and code
-func (c *Client) GetChild(instanceID, name, code string) (m Model, err error) {
+func (c *Client) GetChild(ctx context.Context, instanceID, name, code string) (m Model, err error) {
 	path := fmt.Sprintf("/hierarchies/%s/%s/%s", instanceID, name, code)
 
-	clientlog.Do(context.Background(), "retrieving hierarchy", service, path, log.Data{
+	clientlog.Do(ctx, "retrieving hierarchy", service, path, log.Data{
 		"method":      "GET",
 		"instance_id": instanceID,
 		"dimension":   name,
 		"code":        code,
 	})
 
-	return c.getHierarchy(path)
+	return c.getHierarchy(path, ctx)
 }
 
-func (c *Client) getHierarchy(path string) (m Model, err error) {
-	resp, err := c.cli.Get(c.url + path)
+func (c *Client) getHierarchy(path string, ctx context.Context) (m Model, err error) {
```

Comment:
I am personally not a fan of _named_ return parameters - I think it can be confusing and you can end up with defects because variables have been shadowed and its really hard to spot. 

I won't make you make this change to approve the request but I think it would be better if we did.